src/app/shared/components/header/header.component.ts

```diff
@@ -8,24 +11,28 @@ import { AuthenticationService } from '../../services/authentication.service';
   styleUrls: ['./header.component.scss'],
 })
 
-export class HeaderComponent implements OnInit, OnDestroy {
+export class HeaderComponent implements OnInit {
   userName: string;
   public userData$: Subscription;
   public userData = null;
 
-  constructor( public authenticationService: AuthenticationService ) { }
+  getState: Observable<any>;
 
-  ngOnInit(): void {
-    this.userData$ = this.authenticationService.userData$.subscribe(
-      userData$ => {
-        this.userData = userData$;
-        this.userName = `${this.userData?.name.first} ${this.userData?.name.last}`;
-      },
-    );
+  constructor(
+    public authenticationService: AuthenticationService,
+    private store: Store<AppState>,
+  ) {
+    this.getState = this.store.select(selectUser);
   }
 
-  ngOnDestroy(): void {
-    this.userData$.unsubscribe();
+  ngOnInit(): void {
+    this.getState.subscribe((state) => {
```

Comment:
Create selector that returns userName and subscribe on this selector in declaration section. Remove subscription from life cycle hook.